build.android

```diff
@@ -39,7 +39,7 @@ pushd ${target_dir}
 #       the great change of file system permission in Android 7
 cmake "$@" \
     -DCMAKE_TOOLCHAIN_FILE=${NDK_ROOT}/build/cmake/android.toolchain.cmake  \
-    -DANDROID_NDK=${NDK_ROOT}                                               \
+    -DANDROID_NDK_HOME=${NDK_ROOT}                                               \
```

Comment:
I was confused by this doc, https://github.com/android/ndk-samples/wiki/Configure-NDK-Path#android--ndk--home. It says the ANDROID_NDK_HOME is deprecated while it also says it's useful in CI pipeline.

Reply:
I don't think we need this at all. android.toolchain.cmake will set it correctly, and I didn't find anywhere in our files that it's used.